### PR TITLE
Switch int types to int64, allow use of 64 bits

### PR DESCRIPTION
**Pull request by pbiggar**

### CODE REVIEW

**Comment by pbiggar** [comment]
I think I'd like to ensure we have testing on this before merging.

**Comment by pbiggar** [comment]
tests:

- negative ints (also floats from #4206)
- encoder/decoder roundtrip
- clipboard tests

**Review by StachuDotNet — APPROVED** [review]

File: client/src/core/ProgramTypes.res [hunk 1]
```diff
@@ -76,12 +75,9 @@ module Expr = {
     | Rail
     | NoRail
 
-  // CLEANUP: move comments to LibExecution.ProgramTypes
   @ppx.deriving(show({with_path: false}))
   type rec t =
-    // ints in ReScript only support 32 bit ints but we want 63 bit int
```
[the remark this answers is not in the document]
Comment (pbiggar) [reply client/src/core/ProgramTypes.res:82]: maybe I wasn't aware of int64? I'm not sure

[the remark this answers is not in the document]
Comment (StachuDotNet) [reply client/src/core/ProgramTypes.res:82]: Cool, was just curious

**Pull request merged**